File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/eventuser/EventUserDTO.java
package es.jocasolo.competitiveeventsapi.dto.eventuser;

import java.io.Serializable;
import java.util.Date;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.enums.eventuser.EventUserPrivilegeType;
import es.jocasolo.competitiveeventsapi.enums.eventuser.EventUserStatusType;
import lombok.Getter;
import lombok.Setter;

public class EventUserDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private String userId;

	@Getter
	@Setter
	private String eventId;

	@Getter
	@Setter
	private Date incorporationDate;

	@Getter
	@Setter
	private Date lastStatusDate;

	@Getter
	@Setter
	private EventUserPrivilegeType privilege;

	@Getter
	@Setter
	private EventUserStatusType status;

	@Override
	public String toString() {
		return String.format("EventUserDTO [user=%s, event=%s]", userId, eventId);
	}
}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/eventuser/EventUserDTO.kt
package es.jocasolo.competitiveeventsapp.dto.eventuser

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.BackStackEntryDTO
import es.jocasolo.competitiveeventsapp.enums.eventuser.EventUserPrivilegeType
import es.jocasolo.competitiveeventsapp.enums.eventuser.EventUserStatusType
import java.util.*

class EventUserDTO(
        @SerializedName("userId") var userId: String,
        @SerializedName("eventId") var eventId: String? = null,
        @SerializedName("incorporationDate") var incorporationDate: Date? = null,
        @SerializedName("lastStatusDate") var lastStatusDate: Date? = null,
        @SerializedName("privilege") var privilege: EventUserPrivilegeType? = null,
        @SerializedName("status") var status: EventUserStatusType? = null,
) : BackStackEntryDTO() {

}